### PR TITLE
Rewrite check for admin

### PR DESCRIPTION
**Pull request by narph**
## What does this PR do?

Re writes the HasRoot function.
The token.IsMember(sid) returns TRUE if the caller's process is a member of the Administrators local group. Caller is NOT
expected to be impersonating anyone and is expected to be able to open its own process and process token. In order for the process to "Run as administrator" the user must be in the BUILTIN\Administrators Group.

## Why is it important?

In order to install the elastic agent we are checking for Admin by  accessing "\\.\PHYSICALDRIVE0".
This will not work on all setups since the physical drive might be missing, ex testing with windows containers,  kubelet startup.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

### CODE REVIEW

File: x-pack/elastic-agent/pkg/agent/install/root_windows_test.go [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+import (
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
+
+func TestHasRoot(t *testing.T) {
+	t.Run("check if user is admin", func(t *testing.T) {
+		result, err := HasRoot()
+		assert.NoError(t, err)
+		assert.True(t, result)
```
Comment (blakerouse) [inline x-pack/elastic-agent/pkg/agent/install/root_windows_test.go:18]: Does this mean the tests run as root? I don't think that is always the case.

I run the tests locally sometimes, and they would not be ran as root.

**Review by blakerouse — CHANGES_REQUESTED** [review]
I like this change. I think we originally had something similar, but had to remove it because it didn't work as expected. Hopefully this does not have the same issue (which I can't even remember).

Just have an issue with the test, I think that might need to be fixed.

File: x-pack/elastic-agent/pkg/agent/install/root_windows_test.go (hunk 1 above)
Comment (narph) [reply x-pack/elastic-agent/pkg/agent/install/root_windows_test.go:18]: indeed, so a few options here, either we execute a PowerShell cmd to check and validate if the function returns the same, or we disable the test. Just checked with the automation team, the windows test in CI are always ran under admin so no issues there.

Comment (blakerouse) [reply x-pack/elastic-agent/pkg/agent/install/root_windows_test.go:18]: how about just change it to check the error, and ignore the true or false result?

File: x-pack/elastic-agent/pkg/agent/cmd/uninstall.go [hunk 2]
```diff
@@ -37,8 +37,12 @@ Unless -f is used this command will ask confirmation before performing removal.
...
 
 func uninstallCmd(streams *cli.IOStreams, cmd *cobra.Command, flags *globalFlags, args []string) error {
-	if !install.HasRoot() {
-		return fmt.Errorf("unable to perform uninstall command, not executed with %s permissions", install.PermissionUser)
+	isAdmin, err := install.HasRoot()
+	if err != nil {
+		return fmt.Errorf("unable to perform perform command while checking for administrator rights, %v", err)
+	}
+	if !isAdmin {
+		return fmt.Errorf("unable to perform perform command, not executed with %s permissions", install.PermissionUser)
```
Comment (michalpristas) [inline x-pack/elastic-agent/pkg/agent/cmd/uninstall.go:45]: double perform same above

File: x-pack/elastic-agent/pkg/agent/install/root_windows_test.go (hunk 1 above)
Comment (michalpristas) [reply x-pack/elastic-agent/pkg/agent/install/root_windows_test.go:18]: related issue was here: https://github.com/elastic/beats/pull/21884/files

**Comment by michalpristas** [comment]
related issue is here: https://github.com/elastic/beats/pull/21884/files 
we wll need to recheck this manually on various systems to see if token is not missing

File: x-pack/elastic-agent/pkg/agent/install/root_windows_test.go (hunk 1 above)
Comment (narph) [reply x-pack/elastic-agent/pkg/agent/install/root_windows_test.go]: or that

**Review by michalpristas — APPROVED** [review]
looks correct, please run mage fmt to make sure imports are ordered

File: x-pack/elastic-agent/pkg/agent/cmd/install.go [hunk 3]
```diff
@@ -58,7 +61,7 @@ func installCmd(streams *cli.IOStreams, cmd *cobra.Command, flags *globalFlags,
 
 	// check the lock to ensure that elastic-agent is not already running in this directory
 	locker := application.NewAppLocker(paths.Data(), agentLockFileName)
-	if err := locker.TryLock(); err != nil {
+	if err = locker.TryLock(); err != nil {
```
[the remark this answers is not in the document]
Comment (narph) [reply x-pack/elastic-agent/pkg/agent/cmd/install.go]: made the change

**Pull request merged**